pkg/core/block/header.go

```diff
@@ -250,6 +251,11 @@ func GetExpectedHeaderSize(stateRootInHeader bool, numOfValidators int) int {
 	// n = 1 bytes
 	// 5 for SYSCALL
 	size := expectedHeaderSizeWithEmptyWitness + (1+1+64)*m + 2 + numOfValidators*(1+1+33) + 2 + 5
+	// Genesis block case.
+	if numOfValidators == 0 {
+		// 1 byte for the PUSH1 opcode.
+		size = expectedHeaderSizeWithEmptyWitness + 1
```

Comment:
I need a compatibility test. Create genesis block, serialize it and then compare the resulting size with GetExpectedHeaderSize.

And move `m` and `size` initialization to `else` case.